assets/blueprints/genesis_helper/src/lib.rs

```diff
@@ -0,0 +1,255 @@
+use scrypto::address::Bech32Encoder;
+use scrypto::api::node_modules::metadata::MetadataValue;
+use scrypto::blueprints::account::AccountDepositInput;
+use scrypto::blueprints::epoch_manager::*;
+use scrypto::network::NetworkDefinition;
+use scrypto::prelude::*;
+
+// Important: the types defined here must match those in bootstrap.rs
+type AccountIdx = usize;
+type ResourceIdx = usize;
+type ValidatorIdx = usize;
+
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+pub struct GenesisData {
+    validators: Vec<GenesisValidator>,
+    resources: Vec<GenesisResource>,
+    accounts: Vec<ComponentAddress>,
+    resource_balances: BTreeMap<ResourceIdx, Vec<(AccountIdx, Decimal)>>,
+    xrd_balances: BTreeMap<AccountIdx, Decimal>,
+    stakes: BTreeMap<ValidatorIdx, Vec<(AccountIdx, Decimal)>>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+pub struct GenesisValidator {
+    pub key: EcdsaSecp256k1PublicKey,
+    pub component_address: ComponentAddress,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+pub struct GenesisResource {
+    pub symbol: String,
+    pub name: String,
+    pub description: String,
+    pub url: String,
+    pub icon_url: String,
+    pub address_bytes: [u8; 26],
+    pub owner_with_mint_and_burn_rights: Option<AccountIdx>,
+}
+
+#[blueprint]
+mod genesis_helper {
+    struct GenesisHelper;
+
+    impl GenesisHelper {
+        pub fn init(
+            network: NetworkDefinition,
+            mut genesis_data: GenesisData,
+            mut whole_lotta_xrd: Bucket,
+            validator_owner_token: [u8; 26], // TODO: Clean this up
```

Comment:
not really sure what's the desired clean up here, but carried this comment over from epoch manager :)